pkg/portal/transport/stripe_webhook_handler.go

```diff
@@ -153,16 +153,56 @@ func (h *StripeWebhookHandler) handleCheckoutSessionCompletedEvent(event *libstr
 }
 
 func (h *StripeWebhookHandler) handleCustomerSubscriptionEvent(event *libstripe.CustomerSubscriptionEvent) error {
-	if !event.IsSubscriptionActive() {
-		// If the subscription event is about status that are not active
-		// Ignore it
-		h.Logger.
-			WithField("stripe_subscription_id", event.StripeSubscriptionID).
-			WithField("stripe_subscription_status", event.StripeSubscriptionStatus).
-			Info("unhandled subscription status")
-		return nil
+	// Here is a complete list of subscription status and our corresponding action.
+	// incomplete -> ignore
+	// incomplete_expired -> set checkout to cancelled.
+	// trialing -> ignore
+	// active -> set checkout to subscribed
+	// past_due -> ignore
+	// canceled -> ignore
+	// unpaid -> ignore
+
+	if event.IsSubscriptionActive() {
+		return h.handleActiveSubscriptionEvent(event)
 	}
 
+	if event.IsSubscriptionIncompleteExpired() {
+		return h.handleIncompleteExpiredSubscriptionEvent(event)
+	}
+
+	h.Logger.
+		WithField("stripe_subscription_id", event.StripeSubscriptionID).
+		WithField("stripe_subscription_status", event.StripeSubscriptionStatus).
+		Info("unhandled subscription status")
+	return nil
+}
+
+func (h *StripeWebhookHandler) handleIncompleteExpiredSubscriptionEvent(event *libstripe.CustomerSubscriptionEvent) error {
+	err := h.Database.WithTx(func() error {
+		// Mark checkout session as cancelled
+		err := h.Subscriptions.UpdatedSubscriptionCheckoutStatusToCancelled(
```

Comment:
For diagnostic, is it better to introduce a new status for this case instead of using cancel? Maybe "subscription_incomplete_expired"?

Reply:
Added `expired`